compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java

```diff
@@ -234,15 +293,66 @@ public static String guessOrgName() {
      * Guess package name with valid pattern.
      *
      * @param packageName package name
+     * @param template    template name
      * @return package name
      */
-    public static String guessPkgName(String packageName) {
+    public static String guessPkgName(String packageName, String template) {
         if (!validatePackageName(packageName)) {
-            return packageName.replaceAll("[^a-zA-Z0-9_.]", "_");
+            packageName = packageName.replaceAll("[^a-zA-Z0-9_.]", "_");
+        }
+
+        // if package name is starting with numeric character, prepend `app` / `lib`
+        if (packageName.matches("[0-9].*")) {
+            if (template.equalsIgnoreCase("lib")) {
+                packageName = "lib" + packageName;
+            } else {
+                packageName = "app" + packageName;
+            }
+        }
+
+        // if package name is starting with underscore remove it
+        if (packageName.startsWith("_")) {
+            packageName = removeFirstChar(packageName);
```

Comment:
Without removing the first char let's append app/lib? eg: provided name: _foo, derived name: `app_foo` or `lib_foo`

Reply:
Isn't it better to remove that since we are removing trailing `_` s.
Kept it like this for consistensy.